android-client-sdk/src/main/java/com/devcycle/sdk/android/eventsource/DVCLifecycleCallbacks.kt

```diff
@@ -0,0 +1,70 @@
+package com.devcycle.sdk.android.eventsource
+
+import android.app.Activity
+import android.app.Application
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import timber.log.Timber
+
+class DVCLifecycleCallbacks(eventSource: EventSource, customHandler: Handler? = null) : Application.ActivityLifecycleCallbacks {
+    private val CHECK_DELAY: Long = 800
+
+    private val eventSource = eventSource
+    private val listenerThread: HandlerThread = HandlerThread("DVCForegroundListener")
+    private var handler: Handler? = customHandler
+    private var paused = false
+    private var foreground = true
+    private var check: Runnable? = null
+
+    init {
+        if (handler == null) {
+            listenerThread.start()
+            handler = Handler(listenerThread.looper)
```

Comment:
that's what i had originally but you have to call listenerThread.start() before passing the listenerThread to the Handler or you get a null pointer error :/  so it can't be in the static stuff at the top unfortunately

Reply:
is there a need for a `customHandler` override? You could just remove that from the constructor and then you should be able to make the `handler` property non-nullable since it's being set in the `init` block?

Reply:
oh yeah the customHandler only gets used by the unit tests - nothing has worked to mock the handler / handlerThread / looper constructors to get the tests to run properly without passing in the mocked handler directly. i figured it would be slightly better to have working tests for all the delay logic with this optional parameter added than to have no tests, but also i think it's a bad idea to modify the code to support the tests so i'm okay with removing it